community/sync-gateway/1.0.4/scripts/entrypoint.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+
+# Set up symlinks to log into a 'normal' location.
+mkdir -p /var/log/sync_gateway
+ln -sf /stdout.log /var/log/sync_gateway/sync_gateway_access.log
+ln -sf /stderr.log /var/log/sync_gateway/sync_gateway_error.log
+
+# Run SG and use tee to append stdout and stderr to separate logfiles
+# Process substitution described here: https://stackoverflow.com/a/692407
+exec sync_gateway "$@" > >(tee -a /stdout.log) 2> >(tee -a /stderr.log >&2)
```

Comment:
So in the docker logs, this will interleave stdout and stderr, which I guess isn't the end of the world (and I see no alternative).

The interleaving will be at the line level though and not the character level right?  There's no way to get "messed up lines" where half of the line is from stdout and half is from stderr?  More of a bash/tee question, but I think it's worth proving this out now to save potential headache later.

Reply:
As far as I can tell it's Docker's responsibility to interleave stdout and stderr, so long as I leave those streams intact and separate the behaviour will remain the same.

> By default, docker logs shows the command’s STDOUT and STDERR.
> https://docs.docker.com/config/containers/logging/

Here's a breakdown of what the command is doing on each stream to keep them separate.

#### stdout
Take only stdout (`>`), tee it (which streams back to stdout)
`> >(tee -a /stdout.log)`

#### stderr
Take only stderr (`2>`), tee it but redirect tee's stdout to stderr (`>&2`)
`2> >(tee -a /stderr.log >&2)`